http://bugs.gentoo.org/169477

Revision 2744
Modified Tue Sep 5 19:31:48 2006 UTC (6 months, 1 week ago) by pmandin
Use correct function to free allocated mem

--- branches/SDL-1.2/src/cdrom/linux/SDL_syscdrom.c	2006/07/10 21:03:59	2608
+++ branches/SDL-1.2/src/cdrom/linux/SDL_syscdrom.c	2006/09/05 19:31:48	2744
@@ -218,7 +218,7 @@
 			if ( SDL_strcmp(mnt_type, MNTTYPE_SUPER) == 0 ) {
 				tmp = SDL_strstr(mntent->mnt_opts, "fs=");
 				if ( tmp ) {
-					SDL_free(mnt_type);
+					SDL_stack_free(mnt_type);
 					mnt_type = SDL_strdup(tmp + SDL_strlen("fs="));
 					if ( mnt_type ) {
 						tmp = SDL_strchr(mnt_type, ',');
@@ -229,7 +229,7 @@
 				}
 				tmp = SDL_strstr(mntent->mnt_opts, "dev=");
 				if ( tmp ) {
-					SDL_free(mnt_dev);
+					SDL_stack_free(mnt_dev);
 					mnt_dev = SDL_strdup(tmp + SDL_strlen("dev="));
 					if ( mnt_dev ) {
 						tmp = SDL_strchr(mnt_dev, ',');
